mqtt/handler.go

```diff
@@ -162,32 +162,32 @@ func (h *handler) Publish(c *session.Client, topic *string, payload *[]byte) {
 	// Topics are in the format:
 	// channels/<channel_id>/messages/<subtopic>/.../ct/<content_type>
 
-	channelParts := channelRegExp.FindStringSubmatch(*topic)
-	if len(channelParts) < 2 {
+	subtopicParts := subtopicRegExp.FindStringSubmatch(*topic)
+	if len(subtopicParts) < regExParts {
 		h.logger.Error(LogErrFailedPublish + (ErrMalformedTopic).Error())
 		return
 	}
 
-	chanID := channelParts[1]
-	subtopic := channelParts[2]
-
-	subtopic, err := parseSubtopic(subtopic)
+	subtopic, err := parseSubtopic(subtopicParts[1])
```

Comment:
Rename this method `parseSubtopic` ->  `createSubject` and move it to the messaging/nats package.